dso/model.py

```diff
@@ -381,3 +384,21 @@ class ArtefactMetadata:
     )
     id: int | None = None
     discovery_date: datetime.date | None = None # required for finding specific SLA tracking
+
+    @staticmethod
+    def from_dict(raw: dict):
+        return dacite.from_dict(
+            data_class=ArtefactMetadata,
+            data=raw,
+            config=dacite.Config(
+                type_hooks={
+                    datetime.datetime: dateutil.parser.isoparse,
+                    datetime.date: lambda date: datetime.datetime.fromisoformat(date).date()
+                        if date else None,
```

Comment:
how can `date` be falsy here?

Reply:
Good catch, it should only be falsy if it's `None` but in that case the type hook should not be used 👍🏼